### ISSUE TITLE
NinePatchRects do not display their Textures after certain circumstances

### ISSUE THREAD

**Issue by Nosrick** [godotengine/godot#47503]
**Godot version:**

v3.2.3.stable.mono.official

**OS/device including version:**

Windows 10 20H2
1030 GT, not sure on driver version
GLES3 backend

**Issue description:**

This is a convoluted one.
I am creating NinePatchRects through code, and assigning them Textures based upon their parent's "Element Name". I use a data-driven UI skinning system, so this is why.
I am also loading all of my Textures from file at the start of the game, and distributing them to the skinnable UI components from a central space.
I load PackedScenes representing each state of my game, and add their Instance() to a Control node I call "MainUI".
When I load in a new PackedScene this way, I free the old Instance() node, and remove it from the tree.
This is where it gets strange.
On the second (and subsequent) time a PackedScene that contains the GUI skinning component scripts are loaded in, and skinned, the NinePatchRects do not display their texture. They are visible, properly sized, and modulated with a non-zero alpha colour.
Even stranger, if I do not free the previous screen, the NinePatchRects of the previous screen do not render their Textures, once the new screen is in place, and are not rendering their Textures, either.

Because of another issue, it is not possible to check if the NinePatchRects have any Textures in the Remote Tree view, but it can be verified with a debugger.

I hope this has a simple fix, and it's something I'm doing wrong!

**Steps to reproduce:**
- Run the test project
- Notice that the first screen has a background, and various UI elements. The pink/black square is not part of the UI skinning system.
- Push Space.
- Notice that the new screen that is instantiated, has NinePatchRects, created in the same way, but their Textures do not render.
- Click the "New Game" 'button'
- Notice that it goes back to the previous screen, and there are no Textures being displayed.

For bonus points:
- Go to LoadingState.cs, and switch the commenting on the GetNextState() function
- Run the project
- Notice that this screen has properly displaying Textures in the NinePatchRects.
- Click "New Game".
- Notice that this screen has no Textures displaying in the NinePatchRects. Again, the black/pink square is not part of the skinnable UI system.
- Repeat the same process as the last section, and note that no Textures are displayed.

**Minimal reproduction project:**

[JoyGodot - Trimmed.zip](https://github.com/godotengine/godot/files/6230653/JoyGodot.-.Trimmed.zip)

**Comment by Nosrick** [comment]
Closing this issue because the minimum reproduction project is massive, and I can't seem to trim it down and still reproduce the bug.
